File: java/src/main/java/io/fortellis/developer/adminapiexample/service/ConnectionService.java
package io.fortellis.developer.adminapiexample.service;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import io.fortellis.developer.adminapiexample.dao.ConnectionDao;
import io.fortellis.developer.adminapiexample.models.ActionLink;
import io.fortellis.developer.adminapiexample.models.ActivateRequest;
import io.fortellis.developer.adminapiexample.models.ActivateResponse;
import io.fortellis.developer.adminapiexample.models.DeactivateResponse;

@Service
public class ConnectionService {

    @Autowired
    private ConnectionDao connectionDao;

    public ActivateResponse activateConnection(final ActivateRequest activateRequest) {
        System.out.println("Activating request: " + activateRequest);

        connectionDao.addConnection(activateRequest);

        ActivateResponse activateResponse = new ActivateResponse();

        List<ActionLink> links = new ArrayList<ActionLink>();
        ActionLink link = new ActionLink();
        link.setHref("https://example.com/deactivate");
        link.setRel("deactivate");
        link.setMethod("POST");
        links.add(link);

        activateResponse.setLinks(links);
        return activateResponse;
    }

    public DeactivateResponse deactivateConnection(final String connectionId) {
        System.out.println("Deactiving connection: " + connectionId);

        connectionDao.deleteConnection(connectionId);

        DeactivateResponse deactivateResponse = new DeactivateResponse();

        List<ActionLink> links = new ArrayList<ActionLink>();
        ActionLink link = new ActionLink();
        link.setHref("https://example.com/activate");
        link.setRel("activate");
        link.setMethod("POST");
        links.add(link);

        deactivateResponse.setLinks(links);
        return deactivateResponse;
    }
}

File: dotnet/Services/ConnectionService.cs
using System;
using System.Collections.Generic;
using AdminAPIImplementation.Models;

namespace AdminAPIImplementation.Services
{
    public class ConnectionService
    {
        public static ActivateResponse activateConnection(ActivateRequest activateRequest)
        {
            Console.WriteLine("Activate request received: ", activateRequest.ToString());

            /********************************************/
            /*    TODO - Perisist activation request.   */
            /********************************************/

            ActivateResponse activateResponse = new ActivateResponse();

            List<ActionLink> links = new List<ActionLink>();
            ActionLink link = new ActionLink();
            link.href = "https://example.com/deactivate";
            link.rel = "deactivate";
            link.method = "POST";
            link.title = "title";
            links.Add(link);

            activateResponse.links = links;

            return activateResponse;
        }

        public static DeactivateResponse deactivateConnection(string connectionId)
        {
            Console.WriteLine("Deactivate request received, connectionId:  {0}", connectionId);

            /***********************************************/
            /* TODO - Perform deactivation business logic. */
            /***********************************************/

            DeactivateResponse deactivateResponse = new DeactivateResponse();

            List<ActionLink> links = new List<ActionLink>();
            ActionLink link = new ActionLink();
            link.href = "https://example.com/activate";
            link.rel = "activate";
            link.method = "POST";
            link.title = "title";
            links.Add(link);

            deactivateResponse.links = links;

            return deactivateResponse;
        }
    }
}
